File: src/aside/java/org/taktik/icure/logic/PlaceLogic.java
package org.taktik.icure.logic;

import org.taktik.icure.entities.Place;
import org.taktik.icure.exceptions.DeletionException;

import java.util.List;

public interface PlaceLogic extends EntityPersister<Place, String>{
    Place createPlace(Place place);

    List<String> deletePlace(List<String> ids) throws DeletionException;

    Place getPlace(String place);

    Place modifyPlace(Place place);

}

File: src/main/kotlin/org/taktik/icure/asynclogic/PlaceLogic.kt
package org.taktik.icure.asynclogic

import kotlinx.coroutines.flow.Flow
import org.taktik.couchdb.DocIdentifier
import org.taktik.icure.asyncdao.PlaceDAO
import org.taktik.icure.entities.Place

interface PlaceLogic : EntityPersister<Place, String> {
    suspend fun createPlace(place: Place): Place?
    fun deletePlace(ids: List<String>): Flow<DocIdentifier>

    suspend fun getPlace(place: String): Place?

    suspend fun modifyPlace(place: Place): Place?
    fun getGenericDAO(): PlaceDAO
}
